### PR TITLE
feat: Add submittions types to data model, types, and frontend, update intake state and activity status

### PR DESCRIPTION
**Pull request by wilwong89**
# Description
Updated intake state and activity status values to title case from uppercase
Added new activity status "New"
Set default activity state to "New" on edit on frontend
Added new submission types to data model and types
Make frontend changes to allow editing new submission types
Added validation to frontend for submission types
Added new value "bcOnlineCompleted" to data model
Added checkbox for BC Online Completed to frontend

## Types of changes

New feature (non-breaking change which adds functionality)

Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [X] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: frontend/src/components/submission/SubmissionForm.vue [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <script setup lang="ts">
-import { Form } from 'vee-validate';
+import { Form, ErrorMessage } from 'vee-validate';
```
Comment (kyle1morel) [inline frontend/src/components/submission/SubmissionForm.vue:2]: alphabetize

File: frontend/src/components/submission/SubmissionForm.vue [hunk 2]
```diff
@@ -340,6 +371,40 @@ const onSubmit = (values: any) => {
...
+      />
+      <Checkbox
+        class="lg:col-2 md:col-6 col-12"
+        name="submissionTypes.inapplicable"
+        label="Inapplicable"
+        :disabled="!props.editable"
+      />
+      <div class="col-12 mb-3">
+        <ErrorMessage name="submissionTypes" />
+      </div>
```
Comment (kyle1morel) [inline frontend/src/components/submission/SubmissionForm.vue:407]: Incorrect styling: 
> New set of checkbox fields in all submissions themselves called: Submission type:  (checkboxes, in the middle column, next to the existing checkboxes, with a grey line vertical divider

File: app/src/types/ChefsSubmissionForm.ts [hunk 3]
```diff
@@ -35,4 +36,9 @@ export type ChefsSubmissionForm = {
   user: User | null; // assigned to
   intakeStatus: string | null;
   applicationStatus: string | null;
+  guidance: boolean | null;
+  statusRequest: boolean | null;
+  inquiry: boolean | null;
+  emergencyAssist: boolean | null;
+  inapplicable: boolean | null;
```
Comment (kyle1morel) [inline app/src/types/ChefsSubmissionForm.ts:43]: You made these non nullable in the schema so the app type should not be nullable.

File: frontend/src/components/submission/SubmissionForm.vue [hunk 4]
```diff
@@ -39,9 +40,19 @@ const assigneeOptions: Ref<Array<User>> = ref([props.submission.user]);
 // Default form values
 const initialFormValues: any = {
   ...props.submission,
+  applicationStatus: props.submission.applicationStatus
+    ? props.submission.applicationStatus
+    : APPLICATION_STATUS_LIST.NEW,
```
Comment (kyle1morel) [inline frontend/src/components/submission/SubmissionForm.vue:45]: We should be setting 'New' when the entry is created in the database. This just masks the issue until they manually resave the data.

**Review by kyle1morel — APPROVED** [review]

**Comment by kyle1morel** [comment]
Preemptively approving - approved once the 500 on Submission load is fixed.

File: app/src/db/models/submission.ts [hunk 5]
```diff
@@ -65,7 +66,12 @@ export default {
       notes: input.notes,
       user: input.user?.userId ? { connect: { userId: input.user.userId } } : disconnectRelation,
       intakeStatus: input.intakeStatus,
-      applicationStatus: input.applicationStatus
+      applicationStatus: input.applicationStatus,
+      guidance: input.guidance ? input.guidance : false,
```
Comment (TimCsaky) [inline app/src/db/models/submission.ts:70]: cant these just be `guidance: input.guidance`
the db field defaults to false.. the html form can submit false

File: frontend/src/components/submission/SubmissionForm.vue (hunk 4 above)
Comment (TimCsaky) [inline frontend/src/components/submission/SubmissionForm.vue:45]: i think this can be removed now that youre adding NEW during chefs data import(?)

**Review by TimCsaky — APPROVED** [review]

**Pull request merged**